polynote-spark/src/main/scala/polynote/kernel/KernelListener.scala

```diff
@@ -180,13 +180,25 @@ class KernelListener(taskManager: TaskManager.Service, session: SparkSession, ru
   override def onTaskEnd(taskEnd: SparkListenerTaskEnd): Unit = {
     import taskEnd.stageId
     if (taskEnd.reason == org.apache.spark.Success) {
-      stageTasksCompleted.get(stageId) match {
+      val task = stageTasksCompleted.get(stageId) match {
         case null =>
           stageTasksCompleted.putIfAbsent(stageId, new AtomicInteger(0))
           stageTasksCompleted.get(stageId).incrementAndGet()
+          jobTasksCompleted.get(stageJobIds.get(stageId)).incrementAndGet()
```

Comment:
should probably make sure `jobTasksCompleted.get(stageJobIds.get(stageId))` doesn't return null?

Reply:
Done!